### PR TITLE
UI: `Menu`

### LINKED ISSUE #78: Create UI components with Reach UI or Radix UI
- [x] [Text](https://minimals.cc/components/typography) #81 
- [x] [Snackbar](https://minimals.cc/components/snackbar) #98 
- [x] [Tab](https://minimals.cc/components/tabs) #118
- [x] [Menu](https://minimals.cc/components/menu)
     * #119 
     * #120
- [ ] [Tooltips](https://minimals.cc/components/tooltip) #129 
- [ ] [Chip](https://minimals.cc/components/chip)

### PR DESCRIPTION
**Pull request by vitokhangnguyen**
Resolves #78 

~~Adding comments, feel free to revierw the implementation first (shouldn't be very hectic)~~

## Changelog:

* [x] Dependencies added:
    *  `@reach/menu-button`
* [x] Addingf resueable `Menu` component
* [x] Refactor `UserMenu` to use the new `Menu` module
    * Remove most logic to manually handle accessibility (`useDropdownMenu` hook).
    * `useClickOutside` hook is worth keeping but is modified to receive `ref` instead of returning a new `ref` because in some cases, we might want to pass a `ref` into many hooks (e.g.: we want to apply `useClickOutside` and `useOverflow` on the same `ref`). Therefore, this should be the way to design hook that works with native elements in the future.
    * Simplify rendering logic: `RouteProps` object was a redundant abstraction, embedding links into the `UserMenu` is easily understandable. This helps when we mismatch links and buttons in the menu as well.

## Notes:

* `UserMenu` currently renders 2 separated lists of items because of a potential bug in Reach UI that throws off the focus order of items inside if an item is rendered asynchronously  https://github.com/reach/reach-ui/issues/876

### CODE REVIEW

**Comment by vitokhangnguyen** [comment]
@Andrewnt219  comments added. Ready for review

File: src/modules/layouts/components/UserMenu/HamburgerIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { VFC } from 'react';
```
Comment (Andrewnt219) [inline src/modules/layouts/components/UserMenu/HamburgerIcon.tsx:1]: Should be from iconify
https://icon-sets.iconify.design/eva/menu-fill/

File: src/ui/Menu/Menu.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Children, isValidElement, ReactNode, FC } from 'react';
+import {
+  Menu as ReachMenu,
+  MenuButton as ReachMenuButton,
+  MenuList as ReachMenuList,
+  MenuListProps as ReachMenuListProps,
+} from '@reach/menu-button';
```
Comment (Andrewnt219) [inline src/ui/Menu/Menu.tsx:7]: Can be `import * as ReachMenu`

File: src/ui/Menu/styles.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import tw, { css } from 'twin.macro';
```
Comment (Andrewnt219) [inline src/ui/Menu/styles.tsx:1]: rename to styles.ts

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]
Nice touch on the comments

File: src/ui/Menu/Menu.tsx (hunk 2 above)
Comment (vitokhangnguyen) [reply src/ui/Menu/Menu.tsx:7]: I think this's fine. Doing `ReachMenu.MenuList` or similarly everywhere is just harder to read. And we're doing this in the Tab anyway.

File: src/ui/Menu/styles.tsx (hunk 3 above)
Comment (vitokhangnguyen) [reply src/ui/Menu/styles.tsx:1]: Gonna do the same for `Button`

**Review by Andrewnt219 — APPROVED** [review]
a11y is a real pain huh.

**Comment by vitokhangnguyen** [comment]
> a11y is a real pain huh.

*There's a lib for that*, ppl said... *It's gonna be easy* ppl said...

What do you do if the lib doesn't support your use case or, god forbid, has a bug? :)

**Pull request merged**